ui-v2/app/routes/dc/kv/create.js

```diff
@@ -20,7 +24,14 @@ export default Route.extend(WithKvActions, {
       create: true,
       isLoading: false,
       item: this.item,
-      parent: repo.findBySlug(key, dc),
+      parent: repo.findBySlug(key, dc).catch(e => {
+        if (e.errors && e.errors[0] && e.errors[0].status == '404') {
+          const url = get(transition, 'intent.url');
+          if (url.endsWith('/create')) {
+            return this.transitionTo('dc.kv.folder', key + '/create/');
+          }
+        }
+      }),
```

Comment:
And this is handling the case where they land here via the url being stripped of the trailing slash - do the data fetch, if it 404s, try it as a folder. I guess this could be where users land if something is deleted between fetching a list, and navigating to the detail as well.

Reply:
Yup, so try and get a `create` key from the API, if its not there, the user could have meant `create/` but it was stripped. So try that by forwarding it back using `ember`.

> I guess this could be where users land if something is deleted between fetching a list, and navigating to the detail as well.

Yeah, if that happens, this is just the same as a 404, so like typing any old thing into the URL under `kv`. It will redirect you back up to the top of the `kv` tree.